include/mxnet/ndarray.h

```diff
@@ -608,6 +549,37 @@ class NDArray {
              << "CheckAndAllocAuxData is not intended for kDefaultStorage";
     ptr_->CheckAndAllocAuxData(i, aux_shape);
   }
+
+#if MXNET_USE_MKLDNN == 1
+  /*
+   * This function returns mkldnn::memory with the default primitive_desc.
+   */
+  std::shared_ptr<const mkldnn::memory> GetMKLDNNData() const;
```

Comment:
How often do these GetMKLXXX() get called, returning shared pointers?  Will this affect performance?

Reply:
We're investigating performance issues. We'll see if shared pointers can cause performance issues.

Reply:
we probably don't need shared ptr, since this is called from inside operator and the NDArray itself will not be released during this time.